package/gluon-config-mode-domain-select/luasrc/lib/gluon/config-mode/wizard/0200-domain-select.lua

```diff
@@ -0,0 +1,48 @@
+return function(form, uci)
+	local site = require 'gluon.site'
+	local path = '/lib/gluon/domains'
+	local fs = require 'nixio.fs'
+	local json = require 'jsonc'
+
+	local function get_domain_list()
+		local list = {}
+		for domain_path in io.popen('find ' .. path .. '/ -iname \*.json -print'):lines() do
+			local domain_code = domain_path:match(path .. '/(.*)\.json$')
+			local domain = assert(json.load(domain_path))
+			table.insert(list, {
+				domain_code = domain_code,
+				domain_name = (domain.domain_names or {})[domain_code],
+				hide_domain = domain.hide_domain or False,
+			})
+		end
+		table.sort(list, function(a,b) return a.domain_name < b.domain_name end)
+		return list
+	end
+
+	local s = form:section(Section, nil, translate('gluon-config-mode:domain-select'))
+
+	local o = s:option(ListValue, 'domain', translate('gluon-config-mode:domain'))
+	o.optional = false
+
+	if uci:get_bool('gluon-setup-mode', uci:get_first('gluon-setup-mode','setup_mode'), 'configured')  then
+		local core = uci:get_first('gluon', 'core')
+		o.default = uci:get('gluon', core, 'domain')
+	else
+		o:value('')
+	end
+
+	for _, domain in pairs(get_domain_list()) do
+		if not domain.hide_domain then
+			o:value(domain.domain_code, domain.domain_name)
+		end
+	end
+
+	function o:write(data)
+		-- local core = uci:get_first('gluon', 'core')
+		-- uci:set('gluon', core, 'domain', data )
+		-- os.execute('gluon-reconfigure')
+		fs.writefile('/tmp/new_domain_code', data)
```

Comment:
What is the problem with the commented code, why write a file in `/tmp`?

Also, we use named sections in Gluon, so you can just use `uci:set('gluon', 'core', ...)` without *get_first*.

Reply:
I had some issues with the commented code before. I will test it again.

Reply:
The main issue I see with the bit above is that you set the value, but don't save it afterwards (`uci:save('gluon')`), so gluon-reconfigure does not see the new value yet. Usually, the save is implied by the returned array `return {'gluon'}`, but in this case the save happens too late.

Moving the gluon-reconfigure call to the reboot step is not a bad idea, but I don't like using `/tmp` for it.